contracts/mocks/SafeERC20Helper.sol

```diff
@@ -46,12 +54,20 @@ contract ERC20SucceedingMock is IERC20 {
     return true;
   }
 
+  function increaseAllowance(address, uint256) public returns (bool){
+    return true;
+  }
+
+  function decreaseAllowance(address, uint256) public returns (bool){
+    return true;
+  }
+
   function balanceOf(address) public view returns (uint256) {
     return 0;
   }
 
   function allowance(address, address) public view returns (uint256) {
-    return 0;
+    return 10;  //non-zero allowance 
```

Comment:
What if we instead add a public function to set this value? That way, we can test that `approve` fails whenever one of the old and new allowances aren't zero.

Reply:
Usually `approve` does that. But here we have a bit different case. So we can add a `setAllowance` function which will set a private variable `allowance` and this variable will be returned by function `allowance`. What you think?

Reply:
Yes, that sounds great! :)

Reply:
done